### DISCUSSION TITLE
Smaller AVR-DD parts available to buy

### DISCUSSION THREAD

**Discussion by obdevel** [SpenceKonde/DxCore#282] [Future development discussion]
Or at least to sample in small quantities. 32K parts and not 64K so far.

I was hoping for a tiny 14-pin QFN footprint but that's only available as SOIC, so I'll have to make do with QFN20 which is 3mm sq.

e.g. https://www.microchip.com/en-us/product/AVR32DD20#buy-from-store. Please save some for me ;)

**Comment by SpenceKonde** [comment]
YEAAAAAH BUDDDYYYYYYYYYY The double-D's are here just in time for swimsuit season. 

I got my order in! 

25 of the 20-pin ones, 25 of the 32 pin ones 40x 14-pin onesbecause I have breakout boards kickingaround waiting for them. And I think I got mostly E-spec ones (extended temp ratnge, so you can overclock harder)

**Comment by obdevel** [comment]
I look on microchip direct every couple of weeks and they just appeared at the weekend.

I won't push you for the pinout pictures yet :p but it would be useful to know which pins will be timer/PWM capable, so I can start laying out a board for a project I have in mind. I need four PWM outputs. I also need one interrupt and one digital out, but I guess I can use any free pins for those. I've been waiting months for that 3x3 QFN20.

**Comment by SpenceKonde** [comment]
I was actually doing the dd28 image as we we speak.

Basically.ypu just take a db, eliminate the portmix errata with tcd, and
there are your pwm pins, less the ones on tabs these don't have

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Comment by obdevel** [comment]
Thanks. This project has a very small double-sided PCB (10x15mm, which is why I've been eagerly awaiting this QFN part), so the routing is pretty challenging.

My plan is (flying kinda blind):

Move TCA0 to port A
PA0 and PA1 remain as TXD0 and RXD0 (defaults)
PA2, 3, 4, 5 are four individual 8-bit PWM outputs (analogWrite)
PD7 is a digital input with interrupt
PC2 is a digital output
RESET (PF6) and UPDI (PF7) as usual
5V to VDD and VDDIO2
GND

Any problems ?

I've also designed a breakout board for testing and debugging, and have 10x AVR32DD20 en route from Thailand :)

**Reply by SpenceKonde** [reply]
I see no issues with that.

**Comment by obdevel** [comment]
Anyone else experiencing a delay with delivery for these new parts ? My order for AVR32DD20s I placed 3 weeks ago is sitting with a 'Shipment Delayed - Contact Us' status. The part page says 'can ship on 14-Jul-2022'. I contacted support and they said they're communicating with the factory. No update so far. Hopefully it's just a delay but my paranoia is working overtime, especially as the page also says 'Additional quantities can ship by 10-Aug-2023' !!

**Reply by SpenceKonde** [reply]
It would appear that the cause of that hiccup was that they found a couple of problems on the initial DD's and were able to rev the die before volume production began. They fixed the errata related to the new HV UPDI stuff. 

Unfortunately, the bugs that they fixed are the only errata on the modern AVRs that I would have preferred stay (on the initial release, sending a  series of HV pulses in rapid succession would cause the chip to fail to time out the UPDI HV override, and so that brought the hope of being able to use the doorbell battery method of getting an HV programmer (the timeout is too short to manually time it right, and it looks like the first crop of chips missed the die rev boat, but the later-released parts held back a few weeks to get it. Unlike every part released before the 2-series, the DD's don't have any truly painful errata. I *really* hope we get a DA and DB die rev. And tinyAVR 1 too. They need it. 

Anyway, the other sizes have been dribbling out. There is now only two package/flash size combos not yet on the market which we know we are getting: 
16DD28 in DIP and 16DD32 in VQFN. 
and then there's the  mystery one, the 64DD20 in VQFN. I don't know if that's happening. I'm wondering if their die is too big. Someone needs to sacrifice a 64DD14 by sanding the encapsulation off and measure the die and see if it's in fact too big to fit in a 3mm x 3mm VQFN.
See https://github.com/SpenceKonde/DxCore/blob/master/megaavr/extras/AboutDxSeries.md for a rather longwinded commentary. 
There was also finally enough data for me to figure out the the pricing relationships between the different pincounts and flash sizes, and I found the results very interesting.
